Ginger/GingerCore/Repository/RepositorySerializer.cs

```diff
@@ -935,49 +935,17 @@ public object DeserializeFromFileObj(Type type, string fileName)
             return DeserializeFromFile(type, fileName);
         }
 
-        public string FileExt(Type T)
+        public string FileExt(RepositoryItemBase repositoryItemBase)
         {
             // making it old Ginger style so BF and env can load again... temp
-            return "Ginger." + GetShortType(T);
+            return "Ginger." + repositoryItemBase.GetItemType();
         }
 
         public object DeserializeFromText(Type t, string s, string filePath = "")
         {
             return DeserializeFromTextWithTargetObj(t, s);
         }
 
-        public string GetShortType(Type t)
-        {
-            //Not so much nice clean OO design but due to static on derived limitation, meanwhile it is working solution 
-            // Put here only class which are saved as stand alone to file system
-            // TODO: add interface to classes which are saved as files which will force them to im
-
-            //TODO: more safe to use type of then Full Name - fix it!
-            if (t == typeof(BusinessFlow)) { return "BusinessFlow"; }
-            if (t == typeof(ActivitiesGroup)) { return "ActivitiesGroup"; }
-            if (t == typeof(Activity)) { return "Activity"; }
-            if (t == typeof(ErrorHandler)) { return "Activity"; }
-            if (t == typeof(CleanUpActivity)) { return "Activity"; }
-            if (typeof(Act).IsAssignableFrom(t)) { return "Action"; }
-            if (typeof(VariableBase).IsAssignableFrom(t)) { return "Variable"; }
-            if (typeof(DataSourceBase).IsAssignableFrom(t)) return "DataSource";
-            if (t.FullName == "GingerCore.Agent") return "Agent";
```

Comment:
any idea why some goes by type and some by FullName?

Reply:
Not sure. will cross-check if it makes any difference.

Reply:
Verified that there is no impact of the different ways we are checking type